### PR TITLE
[None][chore] Add Pro aggregate GSM8K CI gate

### PR DESCRIPTION
**Pull request by mingyangHao**
@coderabbitai summary

## Description

  Adds an aggregate GSM8K accuracy CI gate for the DeepSeek-V4-**Pro** deployment path
  (TP=8, EP=8, attention DP, TRTLLM MoE, FP8 KV cache, MTP `max_draft_len=1`, padded CUDA
  graphs, custom DeepSeek-V4 tokenizer), plus a short-token-boundary smoke test. The smoke
  test is designed to deterministically exercise the empty/short-KV decode path that the
  aggregated GSM8K run only hits non-deterministically.

  Two small robustness fixes are included so the tests run cleanly on this path:
  - `_deepseekv4_pro_agg_llm_kwargs`: set `KvCacheConfig(free_gpu_memory_fraction=0.6)` —
    the default (~0.9) leaves no room for padded CUDA-graph capture and OOMs during engine
    init on large-memory GPUs; these tests need very little KV.
  - `_get_llm_vocab_size`: the DeepSeek-V4 tokenizer raises `NotImplementedError` (not
    `AttributeError`) from `vocab_size`/`__len__`/`get_vocab`, so probe each defensively and
    fall back to the model `config.json` `vocab_size`.

  ## Test Coverage

  New tests (registered in `l0_dgx_b200`):
  - `accuracy/test_llm_api_pytorch.py::TestDeepSeekV4Pro::test_gsm8k_full_accuracy`
  - `accuracy/test_llm_api_pytorch.py::TestDeepSeekV4Pro::test_short_token_boundary_smoke`

  Ran locally on 8x Blackwell, DeepSeek-V4-Pro:
  - `test_gsm8k_full_accuracy`: **PASS** — full GSM8K (1319 samples), accuracy **96.32** vs reference floor 96.0 (~28 min).
  - `test_short_token_boundary_smoke`: **PASS** (~5-8 min).

  Negative control — confirms the smoke test catches the bug it guards: built at the commit
  immediately **before** the DeepGEMM paged-MQA-logits OOB fix (#14940; DeepGEMM pin
  `950d31ab`) and ran `test_short_token_boundary_smoke` — it **FAILS** with
  `CUDA_ERROR_ILLEGAL_ADDRESS (700)` during the `bs=32` all-generation warmup (the all-zero
  `context_lens` path), and **PASSES** at the fixed pin (`245dc5d`). So the smoke test is an
  effective regression guard for that IMA.
## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- If PR introduces API changes, an appropriate PR label is added - either `api-compatible` or `api-breaking`. For `api-breaking`, include `BREAKING` in the PR title.
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.

- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

To see a list of available CI bot commands, please comment `/bot help`.

### CODE REVIEW

File: tests/integration/defs/accuracy/test_llm_api_pytorch.py [hunk 1]
```diff
@@ -3828,6 +3872,37 @@ def test_nvfp4_4gpus_online_eplb(self, moe_backend, mtp_nextn):
...
+    MODEL_PATH = f"{llm_models_root()}/DeepSeek-V4-Pro"
+    EXTRA_EVALUATOR_KWARGS = dict(
+        apply_chat_template=True,
+        system_prompt=_DEEPSEEK_V4_GSM8K_SYSTEM_PROMPT,
+    )
+
+    @pytest.mark.skip_less_mpi_world_size(8)
+    def test_gsm8k_full_accuracy(self):
+        with LLM(self.MODEL_PATH, **_deepseekv4_pro_agg_llm_kwargs()) as llm:
+            _sync_cuda()
```
Comment (jiaganc) [inline tests/integration/defs/accuracy/test_llm_api_pytorch.py]: Why we need a sync before running?

Comment (mingyangHao) [reply tests/integration/defs/accuracy/test_llm_api_pytorch.py]: It is indeed unnecessary, removed.

**Review by jieli-matrix — APPROVED** [review]
